seaborn/utils.py

```diff
@@ -889,3 +891,13 @@ def _disable_autolayout():
 def _version_predates(lib: ModuleType, version: str) -> bool:
     """Helper function for checking version compatibility."""
     return Version(lib.__version__) < Version(version)
+
+
+def try_convert_to_pandas(data: object | None) -> pd.DataFrame:
+    if data is None:
+        return None
+    elif isinstance(data, pd.DataFrame):
```

Comment:
Is this guard important? will passing a `pandas.DataFrame` to ` pd.api.interchange.from_dataframe` be costly? I'd expect it to be a no-op without looking closer.

Reply:
you're right, thanks, it is in-fact a no-op

https://github.com/pandas-dev/pandas/blob/360bf218d68c703911731aec58a52b6501b2f4ce/pandas/core/interchange/from_dataframe.py#L48-L49

However, seaborn still supports versions of pandas older than those which support the interchange protocol, so I introduced this to keep it a no-op in such cases